app/models/concerns/fog_extensions/debug/debug_http_instrumentor.rb

```diff
@@ -0,0 +1,67 @@
+module FogExtensions
+  module Debug
+    class DebugHttpInstrumentor
+      class << self
+        attr_accessor :events
+
+        def instrument(name, params = {}, &block)
+          logger = ::Foreman::Logging.logger('crs')
+          if logger.debug?
+            begin
+              params = params.dup
+              if params.key?(:headers) && params[:headers].key?('Authorization')
+                params[:headers] = params[:headers].dup
+                params[:headers]['Authorization'] = 'FILTERED'
+              end
+              if params.key?(:password)
+                params[:password] = 'FILTERED'
+              end
+              logger.debug("--- #{name} ---")
+              if name.include?('.request')
+                query = ''
+                tmp_query = ''
+                if params.key?(:query) && !params[:query].nil?
+                  params[:query].each do |key, value|
+                    tmp_query += "#{key}=#{value}&"
```

Comment:
`params[:query].map { |p| "#{p[0]}=#{p[1]}" }.join("&")` saves the chomp and explicit string building.